### PR TITLE
flopoco: init at 4.1.3

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

FloPoCo is a generator of arithmetic cores (Floating-Point Cores, but not only) for FPGAs (but not only).

https://flopoco.org

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/electronics/flopoco/default.nix [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "flopoco";
+  version = "4.1.3";
+
+  src = fetchFromGitLab {
+    owner = pname;
+    repo = pname;
+    # flopoco-4.1.3 is not tagged on GitLab
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/flopoco/default.nix:26]: https://gitlab.com/flopoco/flopoco/-/issues/9

File: pkgs/applications/science/electronics/flopoco/default.nix [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    sed -i "s/-pg//g" {,src/Apps/TaMaDi/}CMakeLists.txt
+  '';
+
+  nativeBuildInputs = [
+    cmake
+    installShellFiles
+  ];
+
+  buildInputs = [
+    bison
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/flopoco/default.nix:58]: bison is nativeBuildInputs, too. Please set strictDeps = true to better enforce splicing.

File: pkgs/applications/science/electronics/flopoco/default.nix [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    libxml2
+    mpfi
+    mpfr
+    scalp
+    sollya
+    wcpg
+  ];
+
+  installPhase = ''
+    ./flopoco BuildAutocomplete
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/flopoco/default.nix:71]: Is this building or only installing? if it is building please move to buildPhase

File: pkgs/applications/science/electronics/flopoco/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/applications/science/electronics/flopoco/default.nix]: Done.

File: pkgs/applications/science/electronics/flopoco/default.nix (hunk 3 above)
Comment (wegank) [reply pkgs/applications/science/electronics/flopoco/default.nix:73]: Done.

File: pkgs/applications/science/electronics/flopoco/default.nix [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    install -Dm755 flopoco $out/bin/flopoco
+    cp bin* fp* ieee* longacc* $out/bin/
+    installShellCompletion --bash flopoco_autocomplete
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "The FloPoCo arithmetic core generator";
+    homepage = "http://flopoco.org";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/electronics/flopoco/default.nix:88]:
```suggestion
    homepage = "https://flopoco.org/";
```

**Pull request merged**